grpc/src/main/java/io/stargate/grpc/impl/GrpcImpl.java

```diff
@@ -53,11 +61,15 @@ public GrpcImpl(
 
     int port = Integer.getInteger("stargate.grpc.port", 8090);
 
+    executor =
+        Executors.newScheduledThreadPool(
+            EXECUTOR_SIZE, GrpcUtil.getThreadFactory("grpc-stargate-executor", true));
     server =
         NettyServerBuilder.forAddress(new InetSocketAddress(listenAddress, port))
+            .executor(executor)
```

Comment:
Beyond this ticket, injecting our own executor is a good thing: if we don't, it defaults to `Executors.newCachedThreadPool`, which is unbounded.

Reply:
`Executors.newScheduledThreadPool()` is also unbounded unless we explicitly set the maximum pool size.  

`ScheduledThreadPoolExecutor` inherits `ThreadPoolExecutor` which is the same implementation returned by `Executors.newCachedThreadPool()`.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledThreadPoolExecutor.html

Reply:
It actually overrides that behavior. From the class-level javadoc in that same page:

> because it acts as a fixed-sized pool using `corePoolSize` threads and an unbounded queue, adjustments to `maximumPoolSize` have no useful effect

Reply:
Thanks, I missed that.